### PR TITLE
30 contact form

### LINKED ISSUE #30: Create Contact Form
https://www.octoral.com/en/contact/route

image

### PR DESCRIPTION
**Pull request by dave-fink**
Added contact form with translations.

reCaptcha will be added as a separate ticket - need details from customer.

Fix #30

Test URLs:
- After: https://30-contact-form--octoral--aemsites.hlx.live/en/contact/

### CODE REVIEW

File: blocks/form/form.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      if (field.type === 'radio') {
+        if (field.checked) payload[field.name] = field.value;
+      } else if (field.type === 'checkbox') {
+        if (field.checked) payload[field.name] = payload[field.name] ? `${payload[field.name]},${field.value}` : field.value;
+      } else {
+        payload[field.name] = field.value;
+      }
+    }
+  });
+  return payload;
```
Comment (akasjain2) [inline blocks/form/form.js:73]: Should we add a random id/uuid to each form submission? Not sure what is the best practise in EDS around form submission to sheet, so adding this comment for a suggestion.

File: blocks/form/form.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      },
+    });
+    if (response.ok) {
+      // confirmation
+      const formEl = document.querySelector('.form');
+      const thankYou = div({ class: 'thank-you' }, 'Thank you for filling out this form. Your form has been sent.');
+      formEl.replaceWith(thankYou);
+    } else {
+      const error = await response.text();
+      throw new Error(error);
```
Comment (akasjain2) [inline blocks/form/form.js:103]: In case of a failed submission, should we show a message/alert and reset the form?

File: blocks/form/form-fields.js
Comment (meejain) [inline blocks/form/form-fields.js:1]: Can we add a logic to check a valid email address, if possible . Seems like we are able to submit the form without "@" in the email field. The original site on the form is however showing a red flag if there is not @ XXX.com

**Review by meejain — COMMENTED** [review]
Looks great !!! 

Can we add a logic to check a valid email address, if possible . Seems like we are able to submit the form without "@" in the email field. The original site on the form is however showing a red flag if there is not @ XXX.com

**Comment by dave-fink** [comment]
@akasjain2, @meejain Thank you both for your feedback! I'll incorporate both changes!

**Review by akasjain2 — APPROVED** [review]
LGTM. Few linting errors to address.

**Pull request merged**